Bowling/Bowling/Score.cs

```diff
@@ -0,0 +1,167 @@
+﻿using NUnit.Framework.Constraints;
+using NUnit.Framework.Internal.Execution;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+namespace BowlingApp
+{
+    public class Score
+    {
+        private int _frameNumberTotal;
+
+        public Score()
+        {
+            _frameNumberTotal = 10;
+        }
+
+        public List<Frame> CreateScoreCard(string bowlingMarks)
+        {
+            var frameNumberWithBonusRoll = 10;
+            var frameNumber = 1;
+            List<Frame> framesCollection = new List<Frame>();
+            ArrayList bowlingMarksByFrame = new ArrayList(OrganizeBowlingMarksByFrame(bowlingMarks));
+
+            foreach (var marks in bowlingMarksByFrame)
+            {
+                Frame frame = new Frame();
+
+                if (frameNumber != frameNumberWithBonusRoll)
+                {
+                    FrameNoBonusRollFactory frameNoBonusRoll = new FrameNoBonusRollFactory();
+                    frame = frameNoBonusRoll.Create(frameNumber, marks.ToString());
+                }
+                else
+                {
+                    FrameWithBonusRollFactory frameWithBonusRoll = new FrameWithBonusRollFactory();
+                    frame = frameWithBonusRoll.Create(marks.ToString());
+                }
+
+                framesCollection.Add(frame);
+                frameNumber++;
+            }
+            
+            return framesCollection;
+        }
+
+        public int CalculateScoreCardTotals(List<Frame> framesCollection)
+        {
+            List<Frame> updatedFramesCollection = UpdateFrameScoresForStrikesAndSpares(framesCollection);
+            var grandTotal = CalculateFramesGrandTotal(updatedFramesCollection);
+
+            return grandTotal;
+        }
+
+        private ArrayList OrganizeBowlingMarksByFrame(string bowlingMarks)
```

Comment:
What if we simplify this a bit...my syntax may be off a bit, so copy-paste may not be 100% correct:

```
private IEnumerable<string> OrganizeBowlingMarksByFrame(string bowlingMarks)
{
    var frameMarks = String.Empty;
    var bowlMark = String.Empty;
    var frameCount = 1;

    foreach (var mark in bowlingMarks)
    {
        frameMarks += bowlMark;

        if (!OnLastFrame(frameCount))
        {
            if (!IsFrameCompleted(bowlMark, frameMarks))
            {					
                yield frameMarks;
				
                frameMarks = String.Empty;
                frameCount++;
            }
        }
    }
}

private boolean OnLastFrame(int frameCount) => frameCount == _frameNumberTotal;
private boolean IsFrameCompleted(string bowlMark, string frameMarks) => bowlMark.Equals(RollMarks.strikeMark) || frameMarks.Length == 2;
```

Reply:
Updated OrganizeBowlingMarksByFrame to use yield return as well.